frontend/dockerfile/dockerfile2llb/convert.go

```diff
@@ -1867,6 +1879,25 @@ func addReachableStages(s *dispatchState, stages map[*dispatchState]struct{}) {
 	}
 }
 
+func validateCopySourcePath(excludePatterns, paths []string, location []parser.Range, lint *linter.Linter) error {
+	matcher, err := patternmatcher.New(excludePatterns)
+	if err != nil {
+		return err
+	}
+	for _, src := range paths {
+
+		ok, err := matcher.MatchesOrParentMatches(src)
```

Comment:
I'm not exactly sure if the should be matching parents in here, but I might be missing something.

Reply:
I initially looked to use `Matches` rather than `MatchesOrParentMatches`, however in the comments it was noted:

```
// Deprecated: This implementation is buggy (it only checks a single parent dir
// against the pattern) and will be removed soon. Use either
// MatchesOrParentMatches or MatchesUsingParentResults instead.
```

Which prompted me to this.

Reply:
It's the same dispatch function, depends on if the `source` is build context or not.

It also looks that currently this comparison is for all possible input sources, including for example URLs and Git paths that do not match against dockerignore.

Additionally there are also remote builds that do not use `.dockerignore`. Only paths going to `llb.Local` should be checked.

Reply:
Maybe it was deprecated because the usual transfer methods always need to send/check the parents as well. @aaronlehmann

Reply:
We didn't want to change the semantics of the existing method, but those semantics were wrong.